OpenSim/Common/Component.h

```diff
@@ -2686,26 +2696,26 @@ void ComponentListIterator<T>::advanceToNextValidComponent() {
 template<class C>
 void Connector<C>::findAndConnect(const Component& root) {
  
-    const std::string& path = getConnecteeName();
+    ComponentPath path(getConnecteeName());
     const C* comp = nullptr;
 
     try {
-        if (path[0] == '/') { //absolute path name
-            comp =  &root.template getComponent<C>(path);
+        if (path.isAbsolute()) {
+            comp =  &root.template getComponent<C>(path.getString());
```

Comment:
I like that idea but it could also prove verbose, especially since `C` could be `Millard2012EquilibriumMuscle`. Although it is good to minimize redundancy, I would be okay with an overload; one that takes a string and converts to a `ComponentPath` internally, and one that takes a `ComponentPath`.